src/item_contents.cpp

```diff
@@ -347,21 +348,38 @@ void item_contents::heat_up()
     }
 }
 
-int item_contents::ammo_consume( int qty )
+int item_contents::ammo_consume( int qty, const tripoint &pos )
 {
     for( item_pocket &pocket : contents ) {
-        if( !pocket.is_type( item_pocket::pocket_type::MAGAZINE ) ) {
-            continue;
+        if( pocket.is_type( item_pocket::pocket_type::MAGAZINE_WELL ) ) {
+            // we are assuming only one magazine per well
+            if( pocket.empty() ) {
+                return 0;
+            }
+            // assuming only one mag
+            item &mag = pocket.front();
+            const int res = mag.ammo_consume( qty, pos );
+            if( res && mag.ammo_remaining() == 0 ) {
+                if( mag.has_flag( "MAG_DESTROY" ) ) {
+                    pocket.remove_item( mag );
+                } else if( mag.has_flag( "MAG_EJECT" ) ) {
+                    g->m.add_item( pos, mag );
+                    pocket.remove_item( mag );
+                }
+            }
+            qty -= res;
+        } else if( pocket.is_type( item_pocket::pocket_type::MAGAZINE ) ) {
+
+            qty = pocket.ammo_consume( qty );
         }
-        qty = pocket.ammo_consume( qty );
     }
     return qty;
 }
 
 item *item_contents::magazine_current()
```

Comment:
Should this return an `item_location`?

Reply:
most functions that take item * or item & as parameters and return item * or item & really should imo. it's not important enough to do that in this PR though.